uvicorn/supervisors/multiprocess.py

```diff
@@ -23,11 +23,32 @@ def __init__(self, config, target, sockets):
         self.processes = []
         self.should_exit = threading.Event()
         self.pid = os.getpid()
+        self.child_pids = []
 
     def signal_handler(self, sig, frame):
         """
         A signal handler that is registered with the parent process.
         """
+        logger.info(f"Handling signal: {sig}")
+        for child_pid in self.child_pids:
+            logger.debug(f"Attempt at killing child PID {child_pid}")
+            try:
+                os.kill(child_pid, signal.SIGINT)
+                (pid, status) = os.waitpid(child_pid, 0)
+                if pid == child_pid:
+                    logger.debug(f"{pid}: {status}")
+                    if os.WIFEXITED(status):
+                        logger.debug(
+                            "process returning status exited via the exit() system call"
+                        )
+                    elif os.WIFSIGNALED(status):
+                        logger.debug(
+                            "process returning status was terminated by a signal"
+                        )
+                    elif os.WIFSTOPPED(status):
+                        logger.debug("process returning status was stopped")
```

Comment:
Do we _need_ all this debug information? I think we should keep this as lightweight as possible.